contract/test/test-sell-concert-tickets-contract.js

```diff
@@ -80,40 +103,55 @@ test('Start the contract', async t => {
  * @param {ZoeService} zoe
  * @param {ERef<import('@agoric/zoe/src/zoeService/utils').Instance<AssetContractFn>} instance
  * @param {Purse} purse
- * @param {string[]} choices
+ * @param {[string, NatValue][]} choices
+ * @param {boolean} successfulTrade
  */
-const alice = async (t, zoe, instance, purse, choices = ['map', 'scroll']) => {
+const alice = async (
+  t,
+  zoe,
+  instance,
+  purse,
+  choices = [
+    ['frontRow', 1n],
+    ['middleRow', 1n],
+  ],
+  successfulTrade = true,
+) => {
   const publicFacet = E(zoe).getPublicFacet(instance);
   // @ts-expect-error Promise<Instance> seems to work
   const terms = await E(zoe).getTerms(instance);
-  const { issuers, brands, tradePrice } = terms;
+  const { issuers, brands } = terms;
 
-  const choiceBag = makeCopyBag(choices.map(name => [name, 1n]));
+  const choiceBag = makeCopyBag(choices);
+  const totalPrice = bagPrice(choiceBag, terms.inventory);
   const proposal = {
-    give: { Price: tradePrice },
-    want: { Items: AmountMath.make(brands.Item, choiceBag) },
+    give: { Price: totalPrice },
+    want: { Tickets: AmountMath.make(brands.Ticket, choiceBag) },
   };
-  const pmt = await E(purse).withdraw(tradePrice);
+  const pmt = await E(purse).withdraw(totalPrice);
   t.log('Alice gives', proposal.give);
-  // #endregion makeProposal
 
   const toTrade = E(publicFacet).makeTradeInvitation();
 
   const seat = E(zoe).offer(toTrade, proposal, { Price: pmt });
-  const items = await E(seat).getPayout('Items');
+  const tickets = await E(seat).getPayout('Tickets');
```

Comment:
something like this... this probably doesn't return early enough in some cases...

```suggestion
  const result = await E(seat)
    .getOfferResult()
    .catch(err => {
      if (!successfulTrade) {
        t.pass();
      } else {
        t.fail(err.message);
      }
    });
  t.log('result', result);
  const tickets = await E(seat).getPayout('Tickets');
```